samples/nrf9160/modem_trace_flash/src/main.c

```diff
@@ -33,45 +25,15 @@ static void on_cfun(enum lte_lc_func_mode mode, void *context)
 	LOG_INF("LTE mode changed to %d\n", mode);
 }
 
-static void print_uart1(char *buf, int len)
-{
-	if (!device_is_ready(uart_dev)) {
-		LOG_ERR("uart1 device not found/ready!");
-		return;
-	}
-	for (int i = 0; i < len; i++) {
-		uart_poll_out(uart_dev, buf[i]);
-	}
-}
+
 
 static void print_traces(void)
 {
-	uint8_t read_buf[READ_BUF_SIZE];
-	int ret;
-	size_t read_offset = 0;
-
-	ret = nrf_modem_lib_trace_data_size();
-	printk("Reading out %d bytes of trace data\n", ret);
-
-	/* Read out the trace data from flash */
-	while (ret > 0) {
-		ret = nrf_modem_lib_trace_read(read_buf, READ_BUF_SIZE);
-		if (ret < 0) {
-			if (ret == -ENODATA) {
-				break;
-			}
-			LOG_ERR("Error reading modem traces: %d", ret);
-			break;
-		}
-		if (ret == 0) {
-			LOG_DBG("No more traces to read from flash");
-			break;
-		}
-
-		read_offset += ret;
-		print_uart1(read_buf, ret);
+	if (IS_ENABLED(MODEM_TRACE_FLASH_SAMPLE_UART)) {
+		print_traces();
```

Comment:
Please use a different name for this function. Now it looks like it calls itself.